extensions/EntityExtension/src/main/kotlin/com/typewritermc/entity/entries/activity/PlayerCloseByActivity.kt

```diff
@@ -51,8 +57,14 @@ class PlayerCloseByActivityEntry(
         context: ActivityContext,
         currentLocation: PositionProperty
     ): EntityActivity<in ActivityContext> {
+        if (activationRange.isPresent && activationRange.get() >= range) {
+            logger.warning("Activation range (${activationRange.get()}) must be smaller than the main range ($range)")
+            activationRange = Optional.empty()
```

Comment:
This should not modify the original field as it violates the internal expectations that entries don't change.

Instead, have the if statement return the value. You can likely use the Optional functions to achieve this more elegantly.